src/app/view/components/dot-large-message-display/services/dot-large-message-display.service.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { DotLargeMessageDisplayService } from './dot-large-message-display.service';
+import { DotcmsEventsServiceMock } from '@tests/dotcms-events-service.mock';
+import { DOTTestBed } from '@tests/dot-test-bed';
+import { DotcmsEventsService } from 'dotcms-js';
+
+describe('DotLargeMessageDisplayService', () => {
+    const mockDotcmsEventsService: DotcmsEventsServiceMock = new DotcmsEventsServiceMock();
+    let dotLargeMessageDisplayService;
+
+    const message: any = {
+        data: {
+            code: { lang: 'eng', content: 'Code Test' },
+            width: '100',
+            body: 'Body Test',
+            title: 'testTitle',
+            lang: 'eng',
+            height: '200'
+        }
+    };
+
+    beforeEach(() => {
+        const injector = DOTTestBed.resolveAndCreate([
+            { provide: DotcmsEventsService, useValue: mockDotcmsEventsService },
+            DotLargeMessageDisplayService
+        ]);
+
+        dotLargeMessageDisplayService = injector.get(DotLargeMessageDisplayService);
+    });
+
+    it('should emit a message', (done) => {
+        mockDotcmsEventsService.triggerSubscribeTo('LARGE_MESSAGE', message);
+        dotLargeMessageDisplayService.sub().subscribe((msg) => {
+            const { code, width, body, title, lang, height } = message.data;
+            const emittedMsg = {
+                title,
+                height,
+                width,
+                body,
+                code: { lang, content: code }
+            };
+
+            expect(msg).toEqual(emittedMsg);
```

Comment:
Why not `expect(msg).toEqual(message.data);`? Also, is better it you called `messageMock` so we understand at first glance is a mock, it took me a couple of min to get it.

Reply:
Maybe this test is working because of the `done` but there is a problem here, you are triggering the message before subscribe, it should subscribe first and trigger the message after,

Reply:
1) the `message` from the BE comes with a different structure
![image](https://user-images.githubusercontent.com/37185433/54626640-2ff8c500-4a37-11e9-8937-0c0af4ad29a9.png)

Reply:
2) since it's a `BehaviourSubject`, when you first subscribe you get `null` 
https://github.com/dotCMS/core-web/blob/6e876a17e9f323ff597df49b9ff6983c451bc50e/src/app/view/components/dot-large-message-display/services/dot-large-message-display.service.ts#L20

that's why I send the message before the subscribe, so I can compare the data once that I subscribe